From 6398e9f43e81daaa424fd9e58f1acaaeaba62cf4 Mon Sep 17 00:00:00 2001
From: Paul Fariello <pfariello@arkoon.dev>
Date: Mon, 20 Oct 2014 18:25:12 +0200
Subject: [PATCH] Fix build

libluajit requires libm but -lm was set after libluajit and so libm was
discarded under some conditions
---
 lib/haka/CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/haka/CMakeLists.txt b/lib/haka/CMakeLists.txt
index d296a83..1160fda 100644
--- a/lib/haka/CMakeLists.txt
+++ b/lib/haka/CMakeLists.txt
@@ -80,6 +80,11 @@ set_target_properties(libhaka PROPERTIES COMPILE_DEFINITIONS _GNU_SOURCE)
 set_target_properties(libhaka PROPERTIES VERSION ${HAKA_VERSION_MAJOR}.${HAKA_VERSION_MINOR}.${HAKA_VERSION_PATCH}
 	SOVERSION ${HAKA_VERSION_MAJOR})
 
+target_link_libraries(libhaka LINK_PRIVATE  ${LUA_LIBRARIES})
+add_dependencies(libhaka ${LUA_DEPENDENCY})
+include_directories(${LUA_INCLUDE_DIR})
+link_directories(${LUA_LIBRARY_DIR})
+
 target_link_libraries(libhaka LINK_PRIVATE ${DL_LIBRARIES})
 target_link_libraries(libhaka LINK_PRIVATE ${CMAKE_THREAD_LIBS_INIT})
 target_link_libraries(libhaka LINK_PRIVATE ${INIPARSER_LIBRARY})
@@ -87,11 +92,6 @@ target_link_libraries(libhaka LINK_PRIVATE rt)
 target_link_libraries(libhaka LINK_PRIVATE m)
 target_link_libraries(libhaka LINK_PRIVATE ${EDITLINE_LIBRARY})
 
-target_link_libraries(libhaka LINK_PRIVATE  ${LUA_LIBRARIES})
-add_dependencies(libhaka ${LUA_DEPENDENCY})
-include_directories(${LUA_INCLUDE_DIR})
-link_directories(${LUA_LIBRARY_DIR})
-
 set_target_properties(libhaka PROPERTIES OUTPUT_NAME haka)
 
 # Install
